lib/jekyll/filters.rb

```diff
@@ -382,17 +382,10 @@ def as_liquid(item)
       when Array
         item.map { |i| as_liquid(i) }
       else
-        if item.respond_to?(:to_liquid)
-          liquidated = item.to_liquid
-          # prevent infinite recursion for simple types (which return `self`)
-          if liquidated == item
-            item
-          else
-            as_liquid(liquidated)
-          end
-        else
-          item
-        end
+        # prevent infinite recursion for simple types (which return `self`) when
+        #   `item` responds to :to_liquid
+        return item if !item.respond_to?(:to_liquid) || (item.to_liquid == item)
+        as_liquid(item.to_liquid)
```

Comment:
It looks like this is calling `item.to_liquid()` twice, instead of caching the results.

I agree that the two line refactor _looks_ nicer, but it is not so good for performance.

Reply:
Is there a benchmark we could rely on to measure such a change?

Reply:
I’m not sure

Reply:
Created a small benchmark for it: https://gist.github.com/Crunch09/b5d96d730829c6baf395739349ac1c93 (test case is mostly copied form `test_filter.rb`). @pathawks is right, master is clearly quicker.

Reply:
Thank you for the benchmark code, @Crunch09! Always wondered if one could easily measure the impact of a refactor on performance..